databricks/koalas/frame.py

```diff
@@ -9328,34 +9328,57 @@ def filter(self, items=None, like=None, regex=None, axis=None):
             else:
                 raise ValueError("items should be a list-like object.")
             if axis == 0:
-                # TODO: support multi-index here
-                if len(index_scols) != 1:
-                    raise ValueError("Single index must be specified.")
-                col = None
-                for item in items:
-                    if col is None:
-                        col = index_scols[0] == F.lit(item)
-                    else:
-                        col = col | (index_scols[0] == F.lit(item))
+                if len(index_scols) == 1:
+                    col = None
+                    for item in items:
+                        if col is None:
+                            col = index_scols[0] == F.lit(item)
+                        else:
+                            col = col | (index_scols[0] == F.lit(item))
+                elif len(index_scols) > 1:
+                    # for multi-index
+                    cols = []
+                    col = None
+                    for item in items:
+                        for i, element in enumerate(item):
+                            if col is None:
+                                col = index_scols[i] == F.lit(element)
+                            else:
+                                col = col & (index_scols[i] == F.lit(element))
+                        cols.append(col)
+                        col = None
+                    for c in cols:
+                        if col is None:
```

Comment:
Is it correct? Seems `col` is always going to be `None`.

Reply:
Now, it is checked only when the first value is entered in `col`.